File: javascript/src/domain/calculate/subtotal.js
module.exports = class Subtotal {

    constructor(count, sum) {
        this.count = count;
        this.sum = sum;
    }

    add(value) {
        this.sum = this.sum + value;
        this.count++;
        return this;
    }

    getCount() {
        return this.count;
    }

    getSum() {
        return this.sum;
    }
}

File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/calculate/Subtotal.java
package nl.craftsmen.workshops.reactivemeetup.domain.calculate;

public class Subtotal {

	private final int count;
	private final double sum;

	public Subtotal(int count, double sum) {
		this.count = count;
		this.sum = sum;
	}

	public Subtotal add(double value) {
		return new Subtotal(count + 1, sum + value);
	}

	public int getCount() {
		return count;
	}

	public double getSum() {
		return sum;
	}
}
